crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -454,6 +455,49 @@ module dwallet_system::dwallet {
         encryption_key_id: ID,
     }
 
+    /// Owned object that holds all encrypted user shares that were sent to the user,
+    /// and have been verified, re-signed, and re-encrypted by the user.
+    /// `encrypted_user_shares` is a key-value table where the key is the Dwallet ID and the value is the EncryptedUserShare object ID.
+    struct EncryptedUserShares has key {
+        id: UID,
+        encrypted_user_shares: Table<ID, ID>,
+    }
+
+    public fun create_encrypted_user_shares(ctx: &mut TxContext) {
+        let holder = EncryptedUserShares {
+            id: object::new(ctx),
+            encrypted_user_shares: table::new(ctx),
+        };
+        transfer::transfer(holder, tx_context::sender(ctx));
+    }
+
+    public fun save_encrypted_user_share(
+        encrypted_user_shares: &mut EncryptedUserShares,
+        encrypted_user_share: &EncryptedUserShare,
+        encryption_key: &EncryptionKey,
+        ctx: &mut TxContext
+    ) {
+        // TODO: add signature verification
+        let validate_parameters = !table::contains(&encrypted_user_shares.encrypted_user_shares, encrypted_user_share.dwallet_id)
+            && object::id(encryption_key) == encrypted_user_share.encryption_key_id
```

Comment:
```suggestion
```
I think this check is not needed

Reply:
Why not? Don't we want to enforce users to save only user shares encrypted by their own key?

Reply:
Encryptions are not said to be "by a key" but "to a key" - since this key is public and anyone can encrypt to it. You are right, we should have this check.

Reply:
But I am missing another check: that the `EncryptedUserShare` is actually encrypted *by* the user. This doesn't exist here because you didn't base the PR on the branch that includes the signature inside `EncryptedUserShare`. Do that pleas